gap/ClassicalMaximals.gi

```diff
@@ -827,3 +827,25 @@ function(n, q, classes...)
 
     return maximalSubgroups;
 end);
+
+BindGlobal("C3SubgroupsSymplecticGroupGeneric",
+function(n, q)
+    local primeDivisorsOfn, s, result;
+
+    primeDivisorsOfn := PrimeDivisors(n);
+    result := [];
+
+    # symplectic type subgroups
+    for s in primeDivisorsOfn do
+        if IsEvenInt(n / s) then
+            Add(result, SymplecticSemilinearSp(n, q, s));
+        fi;
+    od;
```

Comment:
I think this could be improved somewhat. Let me rephrase what you do: you are looping over all even divisors `d` (= `n/s`) of `n` and then do something with `n/d` (=`n/(n/s)`=`s`). But the even divisors of `n` either don't exist (if `n` is odd) or they are equal to the divisors of `n/2`, each multiplied by 2.

Reply:
Actually, here of course you only look at the *prime* divisors, so basically all you need is to let s iterate over the *odd* prime divisors. So unless I am missing something, one could do:
```suggestion
    if not IsEvenInt(n) then return []; fi;

    primeDivisorsOfn := PrimeDivisors(n);
    RemoveSet(primeDivisorsOfn, 2)
    result := [];

    # symplectic type subgroups
    for s in primeDivisorsOfn do
        Add(result, SymplecticSemilinearSp(n, q, s));
    od;
```
However, my original remark still applies to the C2 case in PR #77.

Reply:
Be careful here: If n is divisible by 4, then s = 2 is still admissible (as n / s = n / 2 is still even) - so you can't just throw out s = 2 entirely. 

Of course, you could add an if-statement saying "throw out s = 2 if n is not divisible by 4, otherwise leave it in", but I think then the new syntax would not only be not much easier or clearer than the original one, but it would also obfuscate the intention behind it all (namely, that we just need to ensure that n / s is even).